worlds/sc2/options.py

```diff
@@ -1093,6 +1092,20 @@ def get_excluded_missions(world: 'SC2World') -> Set[SC2Mission]:
     # Omitting missions not in enabled campaigns
     for campaign in disabled_campaigns:
         excluded_missions = excluded_missions.union(campaign_mission_table[campaign])
+    # Omitting unwanted mission variants
+    if get_option_value(world, "enable_race_swap") == EnableRaceSwapVariants.option_pick_one:
+        swaps = [
+            mission for mission in SC2Mission
+            if mission not in excluded_missions
+            and mission.flags & (MissionFlag.HasRaceSwap|MissionFlag.RaceSwap)
+        ]
+        while len(swaps) > 0:
+            curr = swaps[0]
+            variants = list([mission for mission in swaps if mission.map_file == curr.map_file])
```

Comment:
you're calling list() on something that's already a list and is created on same row

I'd add `variants.sort(key=lambda mission: mission.id)` so the order is ensured to be deterministic before picking a random one by `world.random`